src/control/server/instance_storage_rpc.go

```diff
@@ -156,3 +156,87 @@ func (ei *EngineInstance) StorageFormatNVMe() (cResults proto.NvmeControllerResu
 
 	return
 }
+
+func smdGetHealth(ctx context.Context, ei *EngineInstance, dev *ctlpb.SmdDevice) error {
+	state := dev.Ctrlr.DevState
+	if state == ctlpb.NvmeDevState_NEW {
```

Comment:
Wouldn't it be better to skip if the device is in any state other than NORMAL? e.g. What happens if you try to get the health of an EVICTED or UNPLUGGED device?

Reply:
retrieving the health of an evicted device should succeed and is an intended use case, unplugged devices should not be returned from smd list, I can restrict this check to only retrieve health stats if in evicted or normal state, done